incubator/mariadb/maria.libsonnet

```diff
@@ -0,0 +1,321 @@
+local k = import 'ksonnet.beta.2/k.libsonnet';
+local deployment = k.extensions.v1beta1.deployment;
+
+{
+  parts:: {
+
+    svc(namespace, name, metricsEnabled=true, labels={app:name}, selector={app:name})::
+      local defaults = {
+        serviceType: "ClusterIP"
+        // i could totally hardcode this and skip the defaults object
+      };
+      {
+        apiVersion: "v1",
+        kind: "Service",
+        metadata: {
+          name: name,
+          labels: labels,
+          [if metricsEnabled then "annotations"]: {
+            "prometheus.io/scrape": "true",
+            "prometheus.io/port": "9104",
+          }
+        },
+        spec: {
+          type: defaults.serviceType,
+          ports: [
+            {
+              name: "mysql",
+              port: 3306,
+              targetPort: "mysql",
+            },
+          ] + if metricsEnabled then [
+            {
+              name: "metrics",
+              port: 9104,
+              targetPort: "metrics",
+            },
+          ] else [],
+          selector: selector,
+        },
+      },
+
+    secret(namespace, name, mariaRootPassword, mariadbPassword, labels={app:name},):: {
+      apiVersion: "v1",
+      kind: "Secret",
+      metadata: {
+        name: name,
+        namespace: namespace,
+        labels: labels,
+      },
+      type: "Opaque",
+      data: {
+         "mariadb-root-password":
+         if mariaRootPassword != null then std.base64(mariaRootPassword)
+          else error "Please add a root password",
```

Comment:
These should never be null. You can remove these conditionals.

Reply:
depressing checkmark